karton/classifier/classifier.py

```diff
@@ -139,17 +139,31 @@ def process(self, task: Task) -> None:
             sample_classes += self._classify_yara(task)
 
         filemagic_classification = self._classify_filemagic(task)
-        if filemagic_classification:
+        if filemagic_classification["kind"] is not None:
             sample_classes.append(filemagic_classification)
 
         file_name = sample.name or "sample"
 
         if not sample_classes:
-            self.log.info(
-                "Sample {!r} not recognized (unsupported type)".format(
-                    file_name.encode("utf8")
+            if filemagic_classification["magic"] and filemagic_classification[
+                "magic"
+            ].startswith("data"):
+                self.log.info(
+                    "Sample {!r} (sha256 {}) not recognized (unsupported type), "
+                    "first 50 bytes of content: {!r}".format(
+                        file_name.encode("utf8"), sample.sha256, sample.content[:50]
+                    )
+                )
```

Comment:
I see that `file_name.encode("utf8") is very old mistake, there is no point to encode this into bytes and back to str using repr for logging.